lib/pages/chat/chat.dart

```diff
@@ -1294,20 +1298,37 @@ class ChatController extends State<Chat>
         context: context,
       ),
       onSendTap: () {
-        sendMedia(
-          imagePickerController,
-          room: room,
-          caption: _captionsController.text,
-        );
+        _onSendTap(imagePickerController);
         _captionsController.clear();
       },
-      onCameraPicked: (_) => sendMedia(imagePickerController, room: room),
+      onCameraPicked: (_) => _onSendTap(imagePickerController),
       captionController: _captionsController,
       focusSuggestionController: _focusSuggestionController,
       typeAheadKey: _chatMediaPickerTypeAheadKey,
     );
   }
 
+  void _onSendTap(ImagePickerGridController imagePickerController) {
+    mediaCancelTokenMapNotifier.value = {};
+    final tempMap = <String, CancelToken>{};
+
+    for (final selection in imagePickerController.selectedAssets) {
+      final fileName = selection.asset.title;
+      if (fileName != null) {
+        tempMap[fileName] = CancelToken();
```

Comment:
the key should be `{eventId}_{fileName}`, because user may send the same image 2 times

Reply:
Smart idea. I implemented it